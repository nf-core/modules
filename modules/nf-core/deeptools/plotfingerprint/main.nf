process DEEPTOOLS_PLOTFINGERPRINT {
    tag "$meta.id"
    label 'process_high'

    conda "bioconda::deeptools=3.5.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/deeptools:3.5.1--py_0' :
        'quay.io/biocontainers/deeptools:3.5.1--py_0' }"

    input:
    tuple val(meta), path(bams), path(bais)

    output:
    tuple val(meta), path("*.pdf")          , emit: pdf
    tuple val(meta), path("*.raw.txt")      , emit: matrix
    tuple val(meta), path("*.qcmetrics.txt"), emit: metrics
    path  "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extend   = (meta.single_end && params.fragment_size > 0) ? "--extendReads ${params.fragment_size}" : ''
    """
    plotFingerprint \\
        $args \\
        $extend \\
        --bamfiles ${bams.join(' ')} \\
        --plotFile ${prefix}.plotFingerprint.pdf \\
        --outRawCounts ${prefix}.plotFingerprint.raw.txt \\
        --outQualityMetrics ${prefix}.plotFingerprint.qcmetrics.txt \\
        --numberOfProcessors $task.cpus

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        deeptools: \$(plotFingerprint --version | sed -e "s/plotFingerprint //g")
    END_VERSIONS
    """
}
