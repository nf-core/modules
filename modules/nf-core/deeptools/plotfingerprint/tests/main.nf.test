nextflow_process {

    name "Test Process DEEPTOOLS_PLOTFINGERPRINT"
    script "../main.nf"
    process "DEEPTOOLS_PLOTFINGERPRINT"

    tag "modules"
    tag "modules_nfcore"
    tag "deeptools"
    tag "deeptools/plotfingerprint"

    test("homo_sampiens - bam") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.pdf.get(0).get(1)).name,
                                  file(process.out.matrix.get(0).get(1)).name,
                                  process.out.metrics.collect { file(it[1]).readLines().contains("0.24184576629880325") },                                
                                  process.out.versions)
                                  .match()
                }
            )
        }
    }

    test("homo_sampiens - bam - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }
    }
}
