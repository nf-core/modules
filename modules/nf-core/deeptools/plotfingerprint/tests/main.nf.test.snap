{
    "homo_sampiens - bam": {
        "content": [
            "test.plotFingerprint.pdf",
            "test.plotFingerprint.raw.txt",
            [
                false
            ],
            [
                "versions.yml:md5,2dd872e44f5b7284c080491607d9b42a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-05T10:14:38.559795237"
    },
    "homo_sampiens - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.raw.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.qcmetrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,2dd872e44f5b7284c080491607d9b42a"
                ],
                "matrix": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.raw.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.qcmetrics.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "pdf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.plotFingerprint.pdf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2dd872e44f5b7284c080491607d9b42a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-05T10:14:45.23521809"
    }
}